common/src/main/java/io/netty/util/NetUtil.java

```diff
@@ -348,47 +332,47 @@ public static boolean isIpV6AddressesPreferred() {
     }
 
     private static int getIntValue(char c) {
-        switch (c) {
-            case '0':
-                return 0;
-            case '1':
-                return 1;
-            case '2':
-                return 2;
-            case '3':
-                return 3;
-            case '4':
-                return 4;
-            case '5':
-                return 5;
-            case '6':
-                return 6;
-            case '7':
-                return 7;
-            case '8':
-                return 8;
-            case '9':
-                return 9;
+        if (c >= '0' && c <= '9') {
+            return c - '0';
         }
-
-        c = Character.toLowerCase(c);
-        switch (c) {
-            case 'a':
-                return 10;
-            case 'b':
-                return 11;
-            case 'c':
-                return 12;
-            case 'd':
-                return 13;
-            case 'e':
-                return 14;
-            case 'f':
-                return 15;
+        if (c >= 'A' && c <= 'F') {
+            return c - 'A' + 10;
+        }
+        if (c >= 'a' && c <= 'f') {
+            return c - 'a' + 10;
         }
         return 0;
     }
 
+    private static int decimalDigit(String str, int pos) {
+        return str.charAt(pos) - '0';
+    }
+
+    private static byte ipv4WordToByte(String ip, int from, int to) {
+        int ret = decimalDigit(ip, from);
+        from++;
+        if (from == to) {
+            return (byte) ret;
+        }
+        ret = ret * 10 + decimalDigit(ip, from);
+        from++;
+        if (from == to) {
+            return (byte) ret;
+        }
+        return (byte) (ret * 10 + decimalDigit(ip, from));
+    }
+
+    // visible for tests
+    static byte[] validIpV4ToBytes(String ip) {
+        int i;
+        return new byte[] {
+                ipv4WordToByte(ip, 0, i = ip.indexOf('.', 1)),
+                ipv4WordToByte(ip, i + 1, i = ip.indexOf('.', i + 2)),
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract the assignment out of this expression. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "i". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1854)